File: src/jvm/desig/pattern/visitor/ConsumerBill.java
package jvm.desig.pattern.visitor;

/**
 * 支出账单
 */
public class ConsumerBill implements Bill {
    private String item;
    private double amount;

    ConsumerBill(String item, double amount) {
        this.item = item;
        this.amount = amount;
    }

    @Override
    public void accept(AccountBookView accountBookView) {
        accountBookView.view(this);
    }

    String getItem() {
        return item;
    }

    double getAmount() {
        return amount;
    }
}

File: src/kt/desig/pattern/visitor/ConsumerBill.kt
package kt.desig.pattern.visitor

/**
 * 消费单子
 */
class ConsumerBill : Bill {
    internal var item: String = ""
    internal var amount: Double = 0.0

    internal constructor(item: String, amount: Double) {
        this.amount = amount
        this.item = item
    }

    fun getItem(): String {
        return item
    }

    fun getAmount(): Double {
        return amount
    }

    override fun accept(accountBookView: AccountBookView) {
        accountBookView.view(this)
    }
}
